src/client/activation/activationService.ts

```
@@ -62,19 +76,43 @@ export class ExtensionActivationService implements IExtensionActivationService,
             jedi = true;
         }
 
-        const engineName = jedi ? 'Jedi Python language engine' : 'Microsoft Python language server';
-        this.output.appendLine(`Starting ${engineName}.`);
+        await this.logStartup(jedi);
+
         const activatorName = jedi ? ExtensionActivators.Jedi : ExtensionActivators.DotNet;
         const activator = this.serviceContainer.get<IExtensionActivator>(IExtensionActivator, activatorName);
         this.currentActivator = { jedi, activator };
 
         await activator.activate();
     }
+
     public dispose() {
         if (this.currentActivator) {
             this.currentActivator.activator.deactivate().ignoreErrors();
         }
     }
+
+    // write a descriptive text message to output to help us diagnose user issues.
+    private async logStartup(isJedi: boolean): Promise<void> {
+        let outputLine: string = 'Starting Jedi Python language engine.';
+
+        try {
+            if (!isJedi) {
+                outputLine = 'Starting Microsoft Python language server.';
+                const lsFolderService = this.serviceContainer.get<ILanguageServerFolderService>(ILanguageServerFolderService);
+                const msplCurrentFolder = await lsFolderService.getCurrentLanguageServerDirectory();
+                if (msplCurrentFolder && msplCurrentFolder.version) {
+                    outputLine = `Starting Microsoft Python language server (${msplCurrentFolder.version.raw}).`;
+                }
+            }
+        } catch (failReason) {
+            // do not fail doing this task - log only
+            const log: ILogger = this.serviceContainer.get<ILogger>(ILogger);
+            log.logInformation('Failed to obtain current MPLS version during activation.', failReason);
```

Please remove this line, easier to just add `@log('Starting language server')` as follows:
```typescript
@log('Starting language server')
private async logStartup(isJedi: boolean): Promise<void> {
```

Makes it easier to unit test as well, no need to mock ILogger in tests, etc.

---

**No need to change**., but would prefer if you did, this way we get into the habit of using `@log` decorator, instead of writing code for cross cutting concerns...

---

> Please remove this line, easier to just add `@log('Starting language server')`...

I can, but the intention I have here is to only log when there is an error condition that I'd otherwise like to ignore. I thought that by using an informational log message to record the problem, instead of propagating the error or swallowing it and ignoring it altogether, was a palatable alternative. 

> Makes it easier to unit test as well, no need to mock ILogger in tests, etc.

I agree and I love this new feature! However, I'm using the mocks in my test to ensure that I capture this message as one of my path-coverage attempts.

Question: Can I log only under certain conditions? Or would you prefer if I logged using a different function and then decorated that function instead?